src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -137,10 +152,20 @@ export const NavSearchBar: React.FC<NavSearchBarProps> = ({ venueId }) => {
     if (!searchQuery) return [];
 
     return relatedEvents
-      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .filter((event) => {
+        const isEventRoomEnabled =
+          isDefined(event.room) && event.room !== ""
+            ? isDefined(enabledRelatedRoomsByTitle[event.room])
+            : true;
+
+        return (
+          isEventRoomEnabled && event.name.toLowerCase().includes(searchQuery)
+        );
+      })
```

Comment:
This might be more efficient to extract as a `useMemo`'d `relatedEventsInEnabledRooms` or something like that.. (except for the `event.name.toLowerCase().includes(searchQuery)` part which should stay here)